core/domain/collection_services.py

```diff
@@ -1211,6 +1398,8 @@ def load_demo(collection_id: str) -> None:
 
     yaml_content = utils.get_file_contents(demo_filepath)
 
+    # Ruling out the possibility of bytes for mypy type checking.
+    assert isinstance(yaml_content, str)
     collection = save_new_collection_from_yaml(
```

Comment:
But that means this code will fail, right?

This needs to be rewritten. Also looking at the doctrine of https://github.com/oppia/oppia/blob/develop/core/utils.py#L136 it seems that maybe we should actually implement it correctly and also add overloads there and that should fix this issue.

Reply:
But I have doubts, `assert isinstance(yaml_content, str)` passing all the backend tests, so I think the return value is string but if we look at the method https://github.com/oppia/oppia/blob/22e557ebd3d5394aacd9c30c45698fb30d8e5a6a/core/utils.py#L136
In both the cases `whether raw_bytes is true or false` it always returns bytes, also looking at the code implementation it returns bytes ( because .read() always returns bytes, right ?). Can you provide some more info!!

Reply:
> it seems that maybe we should actually implement it correctly

I'm ok to implement it again but I'm not sure on which argument we differentiate the output values for overloads. Because as mentioned in above comment, the method always returns bytes.

Reply:
When `raw_bytes` is `True` then we should return bytes, otherwise str.

Reply:
Fixed